rls/src/build/cargo.rs

```diff
@@ -451,10 +451,14 @@ impl Executor for RlsExecutor {
         // `env::current_exe()` returns) for the shim.
         let rustc_shim = env::var("RUSTC")
             .ok()
-            .or_else(|| env::current_exe().ok().and_then(|x| x.to_str().map(String::from)))
+            .or_else(|| {
+                // Signal to `main_inner()` (the `current_exe()`) to call our
+                // own rustc shim (`rls_rustc::run()`).
+                cmd.env(crate::RUSTC_SHIM_ENV_VAR_NAME, "1");
```

Comment:
While the docs are super useful :star2: I don't think we should wrap this under `or_else` - if I understand correctly this would break in case of `RUSTC=rls` (without passed env var it will not know to act as a shim anymore)

Reply:
Thanks for catching that error, I hadn't considered that possibility, fixing.